### ISSUE TITLE
error LNK2001: unresolved external symbol starts from windows-2022 (20231002.1.0)

### ISSUE THREAD

**Issue by jcwchen** [actions/runner-images#8574]
### Description

Not sure whether it is really a bug in the image, but the failure (error LNK2001: unresolved external symbol from protobuf) only happened after Windows Server 2022 (20231002.1.0). The base code is same in ONNX and the dependency is almost the same as well. The only significant variance is the image version from GitHub Action (20231002.1.0 failed, but 20230924.1.0 worked). Also the latest Windows Server 2019 works fine. I checked both https://github.com/actions/runner-images/pull/8444 and https://github.com/actions/runner-images/pull/8443, but I didn't notice any crucial change that would possibly cause this unresolved external symbol error.

Therefore, I was wondering is there any significant change from 20230924.1.0 to 20231002.1.0 on Windows Server 2022? Any input is welcome. Thank you for looking into it!

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

CI failure: https://github.com/onnx/onnx/actions/runs/6414801812/job/17415759304.
Issue: https://github.com/onnx/onnx/issues/5659.

### Is it regression?

Version: 20231002.1.0

### Expected behavior

 Build the original project (onnx) successfully

### Actual behavior

Build fails after windows 2022: `20231002.1.0`: `onnx.lib(checker.obj) : error LNK2001: unresolved external symbol "public: int __thiscall google::protobuf::RepeatedField<int>::size(void)const " (?size@?$RepeatedField@H@protobuf@google@@QBEHXZ) [D:\a\onnx\onnx\onnx\.setuptools-cmake-build\onnx_cpp2py_export.vcxproj]`

Build success before windows 2022 `20230924.1.0`.

Also, build always succeed on windows 2019 regardless of versions (even the latest one works)

### Repro steps

Follow the same process to build ONNX from source on a Windows Server 2022 machine: https://github.com/onnx/onnx/blob/5f908a9c433d86c4f1dbf6b963f51daf68daa86e/.github/workflows/release_win.yml#L60-L77.

```
git clone onnx
git submodule update --init --recursive
. .\onnx\workflow_scripts\protobuf\build_protobuf_win.ps1
cd onnx
set ONNX_ML=1
set CMAKE_ARGS="-DONNX_USE_PROTOBUF_SHARED_LIBS=OFF -DONNX_USE_LITE_PROTO=ON"
python -m build --wheel
```

**Comment by sergei-pyshnoi** [comment]
Hello @jcwchen. We will take a look.

**Comment by liqunfu** [comment]
thank you @sergei-pyshnoi ! we are scheduled to release onnx and hit this blocking issue. Please help prioritize this work item for us.

**Comment by sergei-pyshnoi** [comment]
We have deployed new i new image version version for Windows 2022 https://github.com/actions/runner-images/releases/tag/win22%2F20231016.1 . Is problem still persist on new images?

**Comment by jcwchen** [comment]
Thank you @sergei-pyshnoi for looking into issue! @liqunfu and I have just found out there is a new introduced library from `"C:\Program Files\MySQL\MySQL Server 8.0\lib\libprotobuf-lite.lib` which causes Protobuf library conflict. We have fixed our build script by prioritizing our installed Protobuf path instead of the prebuilt one. I will close this issue since the root cause has been identified.

Still, it would be super great if we can have an option to pin image version in the future when running GitHub Action to fast unblock some unexpected failure.
